server/src/main/java/org/apache/cloudstack/ha/HAManagerImpl.java

```diff
@@ -156,19 +156,16 @@ public boolean transitionHAState(final HAConfig.Event event, final HAConfig haCo
             if (result) {
                 final String message = String.format("Transitioned host HA state from:%s to:%s due to event:%s for the host id:%d",
                         currentHAState, nextState, event, haConfig.getResourceId());
-                if (LOG.isTraceEnabled()) {
-                    LOG.trace(message);
-                }
+                LOG.debug(message);
```

Comment:
why do we need to up this level?
also why remove the isEnabled check? I'd rather move the message creation inside such a check.

Reply:
@DaanHoogland About removing the `isEnabled`, if you take a look at the methods (`debug`, `trace`, or others) on [Logger](https://github.com/apache/log4j/blob/trunk/src/main/java/org/apache/log4j/Logger.java) and [Category](https://github.com/apache/log4j/blob/trunk/src/main/java/org/apache/log4j/Category.java), you will notice that they already do the same validation as the `isEnabled` do, the only thing that the `if` would avoid is that in some cases the string to be logged would not be created; In an async web app, like `ACS`, it is not relevant to safe this "cost" by adding more lines to the codebase.

About changing the log level to `debug`, in this context, these messages work better as `debug`; as well, it is quite unusual to call `trace` on `ACS`, so, it is interesting to keep a pattern and use `debug`.

Reply:
@GutoVeronezi you are right they do the check and this is an often returning discussion; they do the check after parameter evaluation which in this case is just the passing of the string. This is why I'd like the evaluation/construction of the String inside the `isEnabled` guarded block as well. no need to perform an expensive string contruction if you're not going to use it.

Reply:
@DaanHoogland  I understand your concern, however it is not like we are running `ACS` on an ENIAC; the cost to build a string (with `java.lang`) is barely notable.    
Besides that, it is better to read a code without 'useless' lines.

Furthermore, if we update `log4j` to `log4j 2` it would not be a problem. =)

Reply:
log4j 3 has my vote, big job though. this is a pattern that finds following and this is what i want to prevent, we need very elaborate Sting construction at times. I wouldn't hassle about info or above, but debug and trace, yes. (also aggree on readible code btw, but isEnabled is very explicit. And also agree on your use of `format()`

Reply:
Well, if the real problem is only the cost of building this string, then I would ask you to take another look at the preview code, because, in this case, the string is already built when it reach the `if` statement, I just took off the `if` and changed it to `debug`.

Reply:
Ah, that's why I asked you to move the string creation inside the if block as well.
```suggestion
                if (LOG.isDebugEnabled()) {
                    final String message = String.format("Transitioned host HA state from:%s to:%s due to event:%s for the host id:%d",
                        currentHAState, nextState, event, haConfig.getResourceId());
                    LOG.debug(message);
                }
```
I see now that the message is used later in the block. I don't like this pattern, but you are right.